### PR TITLE
Add support to use Import Maps with @angular-extensions/elements 

### PR DESCRIPTION
**Pull request by felipeplets**
The scenario covered by this PR is discussed in the following Twitter thread: https://twitter.com/FelipePlets/status/1309815632884576259

The main goal is to enable users in micro front end environments to publish components by changing the Import Map definition and only specifying the Import Map key in this directive to dynamically load the component. 

In order to achieve this in a reliable way SystemJS (a widely used library for such cases) needs to be used. This library is loaded once in such environments and is always available in the `window.System` object. This library adds features and polyfill the currently in progress Import Maps spec. 

More details about Import Maps can be found here: https://github.com/systemjs/systemjs/blob/master/docs/import-maps.md

### CODE REVIEW

**Comment by felipeplets** [comment]
@tomastrajan I'm still working in the Unit Tests for the feature but the code and documentation part is finalized for review, so feel free to check it and add your comments in the code. Hope this is useful for the community.

File: projects/elements/src/lib/lazy-elements/lazy-elements-loader.service.ts [hunk 1]
```diff
@@ -182,6 +206,18 @@ export class LazyElementsLoaderService {
...
     }
   }
+
+  private resolveImportMap(url: string) {
+    const System = (window as any).System;
+    if (System) {
+      url = System.resolve(url);
+    } else {
+      throw new Error(
+        `${LOG_PREFIX} - SystemJS is not loaded and thus '${url}' can't be resolved, *axLazyElement depends on SystemJS for import map feature`
```
Comment (tomastrajan) [inline projects/elements/src/lib/lazy-elements/lazy-elements-loader.service.ts:216]: Would mention that this error is a result of using `importMap` AND not having `SystemJs`

**Comment by tomastrajan** [comment]
HI @felipeplets!

This looks very good, happy about all the adjustments to tests, API docs and everything else!  I left one small comment and then I guess we're good to merge!

**Comment by felipeplets** [comment]
Thank you @tomastrajan! Wasn't expecting such quick review and response!

What about `${LOG_PREFIX} - importMap feature depends on SystemJS library to be globally loaded but none was found. Thus '${url}' can't be resolved` for the error message?

**Comment by tomastrajan** [comment]
> Thank you @tomastrajan! Wasn't expecting such quick review and response!
> 
> What about?
> 
> ```
> throw new Error(
>         `${LOG_PREFIX} - importMap feature depends on SystemJS library to be globally loaded but none was found, thus '${url}' can't be resolved, you should either load SystemJS or remove the importMap flag.`
>       );
> ```

Sounds good!

**Comment by felipeplets** [comment]
I had already changed, so seems all good to be merged.

**Pull request merged**